a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -387,53 +387,32 @@ def execute(self, lb_resource):
 
 class UpdateVRIDForLoadbalancerResource(BaseDatabaseTask):
 
-    def execute(self, lb_resource, vrid, port, subnet):
-        vrid_value = CONF.a10_global.vrid
-        if port:
-            if vrid:
-                try:
-                    self.vrid_repo.update(
-                        db_apis.get_session(),
-                        vrid.id,
-                        vrid_floating_ip=port.fixed_ips[0].ip_address,
-                        vrid_port_id=port.id,
-                        vrid=vrid_value,
-                        subnet_id=subnet.id)
-                    LOG.debug("Successfully updated DB vrid %s entry for loadbalancer resource %s",
-                              vrid.id, lb_resource.id)
-                except Exception as e:
-                    LOG.error("Failed to update vrid %(vrid)s "
-                              "DB entry due to: %(except)s",
-                              {'vrid': vrid.id, 'except': e})
-                    raise e
+    def execute(self, lb_resource, vrid_list):
+        if not vrid_list:
+            # delete all vrids from DB for the lB resource's project.
+            self.vrid_repo.delete(db_apis.get_session(),
+                                  project_id=lb_resource.project_id)
+            LOG.debug("Successfully deleted DB vrid from project %s",
+                      lb_resource.project_id)
+            return
+        for vrid in vrid_list:
+            if self.vrid_repo.exists(db_apis.get_session(), vrid.id):
+                self.vrid_repo.update(
```

Comment:
I see `try-except` has been removed and I guess it was added for "proper reverts stories" in case of failure. Is it alright?

Reply:
Added try-except for delete, update and create Db calls.